doc/user_guide/tools.rst

```diff
@@ -95,27 +97,36 @@ The different subclasses are characterized by three
     :py:meth:`~.signal.BaseSignal.set_signal_origin` changes the signal_origin in place,
     which may result in a :py:class:`~.signal.BaseSignal` subclass transformation.
 
+`dtype`
```

Comment:
Maybe we shouldn't document this in this way here. The `_dtype` attribute is private and the users shouldn't need to know about it. What about simply explaining (using your diagram) that the data dtype affects the subclass assignment and that the preferred way to change the `dtype` is using the `change_dtype` method?